Sources/SwiftKueryORM/Model.swift

```diff
@@ -462,19 +490,38 @@ public extension Model {
   /// Delete all the models matching the QueryParams
   /// - Parameter using: Optional Database to use
   /// - Returns: An optional RequestError
-  static func deleteAll<Q: QueryParams>(using db: Database? = nil, matching queryParams: Q, _ onCompletion: @escaping (RequestError?) -> Void) {
+  static func deleteAll<Q: QueryParams>(using db: Database? = nil, matching queryParams: Q?, _ onCompletion: @escaping (RequestError?) -> Void) {
     var table: Table
-    var filter: Filter
     do {
       table = try Self.getTable()
-      filter = try Self.getFilter(queryParams: queryParams, table: table)
     } catch let error {
       onCompletion(Self.convertError(error))
       return
     }
 
-    let query = Delete(from: table).where(filter)
-    Self.executeQuery(query: query, using: db, onCompletion)
+    var query: Delete = Delete(from: table)
+    var parameters: [Any?]? = nil
+    if let queryParams = queryParams {
+      do {
+        let values: [String: Any] = try QueryEncoder().encode(queryParams)
+        if values.count < 0 {
```

Comment:
How can this ever be less than zero?

Reply:
If all the values in the QueryParams struct are set to nil, it will return `[:]`

Reply:
https://github.com/IBM-Swift/KituraContracts/blob/master/Sources/KituraContracts/CodableQuery/QueryEncoder.swift#L68

Reply:
https://github.com/IBM-Swift/KituraContracts/blob/master/Sources/KituraContracts/CodableQuery/QueryEncoder.swift#L139

Reply:
oh less than 1, my bad

Reply:
Done